src/components/creator/CreatorInputs.tsx

```diff
@@ -0,0 +1,154 @@
+import {
+  FormGroup,
+  NumberInput,
+  Radio,
+  TextArea,
+  TextInput,
+} from '@patternfly/react-core';
+import useChrome from '@redhat-cloud-services/frontend-components/useChrome';
+import React, { useMemo } from 'react';
+import SelectMultiTypeahead from '../SelectMultiTypeahead';
+import { ItemKind, itemKindMeta } from './meta';
+
+export type InputProps<T> = {
+  value: T;
+  onChange: (newValue: T) => void;
+};
+
+export const TypeInput = ({ value, onChange }: InputProps<ItemKind | null>) => {
+  const elementId = `rc-input-type`;
+
+  return (
+    <FormGroup label="Select content type" isRequired fieldId={elementId}>
+      <FormGroup
+        id={elementId}
+        role="radiogroup"
+        aria-label="Select content type"
+      >
+        {Object.entries(itemKindMeta).map(([rawName, meta]) => {
+          const name = rawName as keyof typeof itemKindMeta;
+
+          return (
+            <Radio
+              key={name}
+              id={`cr-input-type-${name}`}
+              name={`cr-input-type`}
+              isChecked={value === name}
+              onChange={(_, isChecked) => {
+                if (isChecked) {
+                  onChange(name);
+                } else if (value === name) {
+                  onChange(null);
+                }
```

Comment:
What happens of none of the conditions are met?

Reply:
* If this radio button is becoming checked, we set the state to the value of this radio button.
* Otherwise, if this radio button is becoming unchecked:
  * If the current state is actually the value of this radio button, set the state to null.
  * If the current state is not the value of this radio button (e.g. because the "becoming checked" event was already processed for another radio button), do nothing.